markdown/developers/modules.md

```diff
@@ -507,22 +507,37 @@ process {
 
 ### Input/output options
 
-1. Input channel declarations MUST be defined for all _possible_ input files (i.e. both required and optional files).
+1. Input channel `path` declarations MUST be defined for all _possible_ input files (i.e. both required and optional files).
 
    - Directly associated auxiliary files to an input file MAY be defined within the same input channel alongside the main input channel (e.g. [BAM and BAI](https://github.com/nf-core/modules/blob/e937c7950af70930d1f34bb961403d9d2aa81c7d/modules/samtools/flagstat/main.nf#L22)).
    - Other generic auxiliary files used across different input files (e.g. common reference sequences) MAY be defined using a dedicated input channel (e.g. [reference files](https://github.com/nf-core/modules/blob/3cabc95d0ed8a5a4e07b8f9b1d1f7ff9a70f61e1/modules/bwa/mem/main.nf#L21-L23)).
 
-2. Named file extensions MUST be emitted for ALL output channels e.g. `path "*.txt", emit: txt`.
+2. Input channel `val` declarations SHOULD be defined for all mandatory non-file inputs that are essential for the functioning of the tool (e.g. parameters, flags etc).
+
+    - Mandatory non-file inputs are options that the tool MUST have to be able to be run.
+    - These non-file inputs are typically booleans or strings, and must be documented as such in the corresponding entry in the `meta.yaml`.
+    - Options, flags, parameters that are _not_ required by the tool to function should NOT be included - rather these can be passed via `ext.args`.
+
+        <details markdown="1">
+        <summary>Rationale</summary>
+        It was decided by a [vote](https://nfcore.slack.com/archives/C043UU89KKQ/p1677581560661679) amongst interested parties within the 2023 Maintainers group on 2023-02-28 to allow non-file mandatory input channels.
+
+        The reasoning behind this was that it is important to have documented (using the existing display on the website) the bare minimum information required for a module to run documented at a nf-core documentation level. It reduces possible risks of entire breakage of modules with future [expected config changes](https://github.com/nextflow-io/nextflow/issues/2723) at a Nextflow level.
+
+        Downsides to this approach are readability (now multiple places must be checked on how to modify a module execution - modules.conf `ext.args`, the module invocation in pipeline code etc.), and reduced user freedom. However it was felt that it was more important for stability in and 'installation' and 'execution' of modules was preferred (e.g. for tools that require position arguments etc.)
```

Comment:
I think the new idea just puts mandatory file inputs on a par with the file inputs. 

Less a case of more places to look for module invocation, more just moving the mandatory non-file inputs to a different bucket in the set I'd have to examine anyway.

Reply:
What I'm trying to say though, is that if someone wants to know which of the tool's flags/parameters are being used - they have to now check multiple places, the entry in modeuls.conf, `TOOL ( ch_input_file, param.non-file)` etc, etc., not _just_ `modules.conf`.

Theortically you wouldn't normally have to check that place (in my experience anyway), but maybe you are more thorough than I am.

Regardless it's one of the speicifc things that @FriederikeHanssen was unhappy about so I thikn it's worth mentioning

Reply:
Just to express myself more clearly (not contesting the inclusion of the passage). I'm not good at guessing what a module needs without examining the channel structure, so:

- if I'm setting up to run a module right now I have to figure out how to provide the full set of:
  -  mandatory inputs, comprised of both file and non-file inputs, spread across channels (for files) and ext.args (for non-files)
  -  non-mandatory inputs, also comprised of file and non-file inputs, spread across channels (for files) and ext.args (for non-files)
- this change simply moves the mandatory non-file inputs from one bucket I already look at (ext.args, which is hard because it's undocumented, I'd have to look at current usage examples), to another (the channels). 

But not looking to lengthen the debate ;-)